EssentialCSharp.Web/Extensions/StringExtensions.cs

```diff
@@ -65,4 +65,17 @@ public static string KeyToHeading(this string str)
     {
         return System.Globalization.CultureInfo.CurrentCulture.TextInfo.ToTitleCase(str.Trim().ToLowerInvariant().Replace('-', ' '));
     }
+
+    public static SiteMapping? Find(this string key, IList<SiteMapping> siteMappings)
+    {
+        key ??= siteMappings[0].Key;
```

Comment:
Is there a case where we are concerned that `key` comes in `null`?

Reply:
Good catch